### PR TITLE
HTTP: Ensure that wp_http_supports() respects custom Requests transports.

### PR DESCRIPTION
**Pull request by dd32**
This ceases using the Deprecated WP_HTTP::_get_first_available_transport() method and calls Requests directly.

Trac ticket: https://core.trac.wordpress.org/ticket/37708

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/http.php [hunk 1]
```diff
@@ -391,7 +389,7 @@ function wp_http_supports( $capabilities = array(), $url = null ) {
 		}
 	}
 
-	return (bool) $http->_get_first_available_transport( $capabilities );
+	return WpOrg\Requests\Requests::has_capabilities( $capabilities );
```
Comment (mukeshpanchal27) [inline src/wp-includes/http.php:408]: Do we needs to mention these change through `@since` annotation?

**Review by peterwilsoncc — APPROVED** [review]
This looks good to me.

As the change is inconsequential to developers using the function, I don't think it _needs_ a since tag update but it also can't hurt.

File: src/wp-includes/http.php (hunk 1 above)
Comment (dd32) [reply src/wp-includes/http.php:408]: I did consider it, but it didn't feel like something that was worthwhile documenting.

From a developer perspective, nothing should change.

Comment (mukeshpanchal27) [reply src/wp-includes/http.php:408]: Goatch.

**Review by mukeshpanchal27 — APPROVED** [review]

**Comment by peterwilsoncc** [comment]
Merged [r58934](https://core.trac.wordpress.org/changeset/58934) / https://github.com/WordPress/wordpress-develop/commit/24e87757cd068b29df98c9f6ae464baaf4659670

**Pull request closed without merging**